react/features/app/actions.js

```diff
@@ -186,18 +164,26 @@ export function appWillUnmount(app) {
  *
  * @param {Object} location - The location URI which specifies the host to load
  * the config.js from.
+ * @private
  * @returns {Promise<Object>}
  */
-function _loadConfig(location: Object) {
-    let protocol = location.protocol.toLowerCase();
+function _loadConfig({ contextRoot, host, protocol, room }) {
+    /* eslint-disable no-param-reassign */
+
+    protocol = protocol.toLowerCase();
 
     // The React Native app supports an app-specific scheme which is sure to not
     // be supported by fetch (or whatever loadConfig utilizes).
     protocol !== 'http:' && protocol !== 'https:' && (protocol = 'https:');
 
     // TDOO userinfo
 
-    return (
-        loadConfig(
-            `${protocol}//${location.host}${location.contextRoot || '/'}`));
+    let url = `${protocol}//${host}${contextRoot || '/'}config.js`;
+
+    // XXX In order to support multiple shards, tell the room to the deployment.
+    room && (url += `?room=${room.toLowerCase()}`);
```

Comment:
This means we'd load the config.js file also on the welcome page, but we currently have no use for it AFAICT. I'd skip it until we really need it, specially given our current inability to generate ETags, which would allow downloading the file altogether, in the common case.